### ISSUE TITLE
Can't login with github "Internal Server Error"

### ISSUE THREAD

**Issue by cvzi** [OpenUserJs/OpenUserJS.org#1037]
I cannot login to my account "cuzi". I am using github for login.
After clicking **Sign in**, I am redirected to `https://openuserjs.org/auth/github/callback/?code=551....&state=C5a....`
That page only says `Internal Server Error`

I tried yesterday and today with Firefox and Chrome and also deleted all cookies.

**Comment by jerone** [comment]
Same issue. Also can't login with Google.

**Comment by Martii** [comment]
Error log

``` sh-session
odules/passport-oauth2/node_modules/oauth/lib/oauth2.js:143:7)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:921:12)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickDomainCallback (node.js:397:17)
```

Reinstalled all deps from scratch in clone and it's still not letting me in as well. Give me a few and I'll see if I can track down but I'm not at dev station right now so I'm limited on development.

**Comment by Martii** [comment]
Additional trace log of:

``` sh-session
2016-10-15 16:28:13.671 +00:00: ESC[31mTokenError: The code passed is incorrect or expired.ESC[39m
2016-10-15 16:28:13.677 +00:00: TokenError: The code passed is incorrect or expired.
    at Strategy.OAuth2Strategy.parseErrorResponse (/home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/lib/strategy.js:320:12)
    at Strategy.OAuth2Strategy._createOAuthError (/home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/lib/strategy.js:367:16)
    at /home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/lib/strategy.js:166:45
    at /home/oujs/OpenUserJS.org1/node_modules/passport-github/lib/strategy.js:77:16
    at /home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/node_modules/oauth/lib/oauth2.js:195:7
    at passBackControl (/home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/node_modules/oauth/lib/oauth2.js:125:9)
    at IncomingMessage.<anonymous> (/home/oujs/OpenUserJS.org1/node_modules/passport-github/node_modules/passport-oauth2/node_modules/oauth/lib/oauth2.js:143:7)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:921:12)
    at nextTickCallbackWith2Args (node.js:442:9)
```

This looks upstream as I haven't changed anything since Friday when I last logged in myself.

**Comment by Martii** [comment]
This had to of been a sub-dep issue... flipped back to the original clone, reinstalled all deps again from scratch and now I can log in. :\ Will retry google auth momentarily...

**Comment by Martii** [comment]
Google works with my moderator account as well.. clean browser tested both for GH and Google.

Hopefully this doesn't recur as it's not our code.

**Comment by Martii** [comment]
Failure on pro with 92bede84eda057fffb2743ee4b88d207f29f9753 but **not dev**. Trying manual hotfix with *async* back to 1.x line...

**Comment by Martii** [comment]
Very strange glitch... logged in and notated [here](https://openuserjs.org/scripts/Marti/RFC_2606%C2%A73_-_Hello%2C_World!/issues/Test_Issue_Creation_on_Pro_1#comment-157d0cb8cc7) with current GH dev of 7e2f6c92c243045d5efc2bff6b923e95a1b51705 running on pro.

**Comment by Martii** [comment]
#### Retest

Different and new error with *async*@2.3.0 on dev:

``` sh-session
$ node app.js
Starting application...
MongoDB connection is opened
Connected to MongoDB v3.2.11
GitHub client authenticated
GET /about 200 20.821 ms - -
GET /favicon.ico 200 69.347 ms - -
GET /redist/npm/octicons/build/font/octicons.css 200 26.646 ms - -
GET /redist/npm/highlight.js/styles/github.css 200 26.291 ms - -
GET /css/common.css 200 24.803 ms - -
GET /redist/npm/font-awesome/css/font-awesome.min.css 200 12.568 ms - -
GET /css/documentPage.css 200 127.575 ms - -
GET /images/favicon.ico 200 752.662 ms - -
GET /redist/npm/bootstrap/dist/js/bootstrap.js 200 753.859 ms - -
GET /redist/npm/jquery/dist/jquery.js 200 753.905 ms - -
GET /less/bootstrap/oujs.css 200 977.702 ms - -
GET /redist/npm/font-awesome/fonts/fontawesome-webfont.woff2?v=4.7.0 200 2.616 ms - 77160
GET /fonts/SquadaOne-Regular.woff2 200 33.706 ms - 5560
GET /register 200 76.549 ms - -
GET /favicon.ico 200 0.724 ms - -
POST /auth/ 302 123.919 ms - 0
[object Object]
Logged in
~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/async.js:61
        fn = function () { throw arg; };
                           ^

TypeError: aSessionIds.indexOf is not a function
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/libs/modifySessions.js:49:23
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/async/dist/async.js:3309:25
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/async/dist/async.js:460:16
    at iteratorCallback (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/async/dist/async.js:1032:13)
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/async/dist/async.js:944:16
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/async/dist/async.js:3306:13
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/libs/modifySessions.js:46:47
    at tryCatcher (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/util.js:16:23)
    at Promise.successAdapter [as _fulfillmentHandler0] (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/nodeify.js:23:30)
    at Promise._settlePromise (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:566:21)
    at Promise._settlePromise0 (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:638:18)
    at Promise._resolveCallback (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:432:57)
    at Promise._settlePromiseFromHandler (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:524:17)
    at Promise._settlePromise (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/promise.js:638:18)
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/bluebird/js/release/nodeback.js:42:21
    at handleCallback (~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/mongoose/node_modules/mongodb/lib/utils.js:120:56)
    at ~/repo/git/OpenUserJS.org/martii/OpenUserJS.org/node_modules/mongoose/node_modules/mongodb/lib/collection.js:1407:5
$
```

**Comment by Martii** [comment]
Closed by 47002bf2e39f1b75d3bf98cc46c9351f6143c958

Helpful when the error thrown upstream is clear. :)
